### ISSUE TITLE
env values in docker-compose files

### ISSUE THREAD

**Issue by Godley** [renovatebot/renovate#4296]
**What would you like Renovate to be able to do?**
We currently use docker-compose extensively for managing our container images. In order to configure the same docker-compose file for local use as well as in a deployment, we use environment variables (which come from a `.env` file). This has native support in docker-compose so you are able to do `image: ${CONTAINER_IMAGE}` without extra hacking (and without telling docker-compose what file to use - it just assumes to look in `.env`)

I'm able to change back to using `image: <whatever>` in order to use renovate for our stack, but there are certain images which are stored in a database (other parts of our infrastructure talk to the database in order to know which container images to launch in other places) and still configured to do that via environment variables.

**Describe the solution you'd like**
It would be nice if renovate could support changes to `.env` files - I notice there's been some consideration to support this on the docker end in other issues. I would imagine it could work quite similarly to the ideas proposed there, e.g
```
CONTAINER_IMAGE=quay.io/repo/someimage:1.1.1 # renovate:docker:quay
```

**Describe alternatives you've considered**
There's not a lot else I can do here - we could move back to having image constraints live within other images (it's a nested container issue) but that was quite complicated and having the image names and tags live in a central location removed a lot of that complexity.
I can't find other tools which have the same wealth of support for this kind of thing as renovate does.

**Additional context**

**Comment by rarkins** [comment]
Depends on #1938

**Comment by nejch** [comment]
I've hacked my way around this issue with a regex manager but it would be nice to have native support. 👍

**Comment by olegkrivtsov** [comment]
I'd like to take this one.

**Comment by olegkrivtsov** [comment]
My plan for implementing this:

1. I guess there will be needed a new manager, e.g. `envfile`? So, I would add it to `lib/manager/envfile`. Take `lib/pyenv` as a base maybe.
2. In the `envfile` manager's `extract.ts`, split the file into lines. For each line, check if it contains `number #renovate: xxxx/yyyyy`. If so, consider this a dependency.

Could you please tell me if this is an acceptable solution?

**Comment by rarkins** [comment]
I think let's go back to "requirements" stage with this one and also put it into a reproduction.

I don't want any new manager - the idea is that people could annotate the docker compose file and not need their own regex manager
